imbi/endpoints/project_score_details.py

```diff
@@ -120,6 +94,41 @@ async def get(self, project_id: int) -> None:
                 } for f in facts if not f.weight]
             })
 
+    async def _gather_enum_facts(
+        self, facts: collections.abc.Sequence[models.ProjectFact]
+    ) -> dict[int, list[dict]]:
+        enum_details = collections.defaultdict[int, list[dict]](list)
```

Comment:
😮 I didn't even know this was legal Python syntax

Reply:
yup ... `defaultdict` became a generic with [PEP-585](https://peps.python.org/pep-0585/) which opened up the syntax in Python 3.9. I think that half of the reason that I am as accepting of Python typing as I am is that it reminds me of C++ 😮 
```c++
typedef std::map<int, std::vector<row_type> > details_type;
details_type enum_details;
```